node/relay_request.js

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var inherits = require('util').inherits;
+
+var EventEmitter = require('./lib/event_emitter');
+var errors = require('./errors');
+
+function RelayRequest(channel, inreq, buildRes) {
+    var self = this;
+    EventEmitter.call(self);
+    self.finishEvent = self.defineEvent('finish');
+
+    self.channel = channel;
+    self.inreq = inreq;
+    self.inres = null;
+    self.outres = null;
+    self.outreq = null;
+    self.buildRes = buildRes;
+}
+inherits(RelayRequest, EventEmitter);
+
+RelayRequest.prototype.createOutRequest = function createOutRequest(host) {
```

Comment:
you added the host parameter.

For cleaning git review please do the move in one commit and the modification of RelayRequest in a second commit.